sled-agent/src/services.rs

```diff
@@ -2,16 +2,39 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-//! Support for miscellaneous services managed by the sled.
+//! Sled-local service management.
+//!
+//! For controlling zone-based storage services, refer to
+//! [crate::storage_manager::StorageManager].
+//!
+//! For controlling virtual machine instances, refer to
+//! [crate::instance_manager::InstanceManager].
+//!
+//! The [ServiceManager] provides separate mechanisms for services where the
+//! "source-of-truth" is Nexus, compared with services where the
+//! "source-of-truth" is the Sled Agent itself. Although we generally prefer to
+//! delegate the decision of "which services run where" to Nexus, there are
+//! situations where the Sled Agent must be capable of autonomously ensuring
+//! that zones execute. For example, the "switch zone" contains services which
+//! should automatically start when a Tofino device is detected, independently
+//! of what other services Nexus wants to have executing on the sled.
+//!
+//! To accomplish this, the following interfaces are exposed:
+//! - [ServiceManager::ensure_persistent] exposes an API to request a set of
+//! services that should persist beyond reboot.
+//! - [ServiceManager::ensure_switch] exposes an API to specifically enable
```

Comment:
I wrote about this in https://github.com/oxidecomputer/rfd/pull/434/commits/228a952060fdf32720d0873e523c00f7dc834b71 , but basically, the switch zone is kinda a "special case", in that it is not controlled by Nexus